src/category_theory/subobject/limits.lean

```diff
@@ -102,25 +106,61 @@ lemma kernel_subobject_factors_iff {W : C} (h : W ⟶ X) :
   kernel_subobject_arrow_comp, comp_zero],
 kernel_subobject_factors f h⟩
 
+instance is_iso_kernel_subobject_zero_arrow : is_iso (kernel_subobject (0 : X ⟶ Y)).arrow :=
+by { rw ←kernel_subobject_arrow, apply_instance, }
+
+@[simp]
+lemma kernel_subobject_zero {A B : C} : kernel_subobject (0 : A ⟶ B) = ⊤ :=
+(is_iso_arrow_iff_eq_top _).mp (by apply_instance)
```

Comment:
I know that it doesn't make a difference in this case, but I'd like to mention that I think that a better way to approach this would be the following:
```lean
@[simp]
lemma kernel_subobject_zero {A B : C} : kernel_subobject (0 : A ⟶ B) = ⊤ :=
(is_iso_iff_mk_eq_top _).mp (by apply_instance)

instance is_iso_kernel_subobject_zero_arrow : is_iso (kernel_subobject (0 : X ⟶ Y)).arrow :=
(is_iso_arrow_iff_eq_top _).mpr kernel_subobject_zero
```
My point is that `kernel_subobject_zero` is a statement that doesn't involve `arrow` at all, so if we have the right API, it should be possible to write its proof entirely in terms of `kernel.ι _` rather than `(kernel_subobject _).arrow`. It may look insignificant here, but I believe that if we are disciplined about this and make sure the API contains all the `mk`-flavored statements we need, then nontrivial proofs can become a lot nicer because we avoid many of the annoying `underlying_iso`s.

Reply:
Ah, okay. I understand.

Reply:
I've used your suggestion.